docker/docker/src/main/java/org/arquillian/cube/docker/impl/docker/DockerClientExecutor.java

```diff
@@ -1023,4 +1043,25 @@ public void onNext(Frame object) {
             }
         }
     }
+
+    private static class StatsLogsResultCallback extends ResultCallbackTemplate<StatsLogsResultCallback, Statistics>{
```

Comment:
This class is already not only super long, but holds too many responsibilities. Can you extract this embedded one to its own file?